pgctl/cli.py

```diff
@@ -238,6 +238,11 @@ def __change_state(self, state):
                     for service in self.all_services
                 )
 
+            # Run the playground-wide "post-start" hook (if it exists) upon every `pgctl start'
+            if state is Start:
+                if all(service.state['state'] == 'ready' for service in self.services):
+                    self._run_playground_wide_hook('post-start')
```

Comment:
I think we may end up with a problem where it's not possible to call some pgctl commands from the post-start hook (since you're already holding the lock in the parent).

I'm okay with keeping this as-is and addressing that if/when it comes up, but I think we should probably make pre-start/post-start/post-stop all either happen with or without the lock for consistency. Currently it's a little odd that post-start has the lock but the other two don't.